### ISSUE TITLE
CPDs of models should be considered as prior in case of Bayesian Learning

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#763]
Currently we pass in the prior probabilities when we call the learning method. Using CPDs as the prior will make the API much cleaner and easier to use.

**Comment by yashu-seth** [comment]
@ankurankan Which learning method do you have in mind?

**Comment by ankurankan** [comment]
@yashu-seth The current API accepts the prior as argument to the fit method for Bayesian Learning. So, I was thinking something like:

1. If there are no CPDs in the network and the user calls the fit method then we should learn using Maximum Likelihood.
2. If there are CPDs in the network and the user calls the fit method, use the Bayesian Learning and use the current CPDs as the prior for the learner.

I feel like this is quite intuitive and will simplify our API.

**Comment by yashu-seth** [comment]
@ankurankan Yes, it might simplify the API. But I as a user would want to fit using different methods available and pick the best suited for my case. And if I want to do this, I won't mind specifying the learning method every time. May be we can default to this behavior and allow specifying learning methods as well.

**Comment by yashu-seth** [comment]
@ankurankan I was looking into the ```BayesianEstimator``` class. The priors given are of three types. But I could not figure out where prior probabilities are incorporated. Can you help me with this? Besides which paper was used to implement the ```BayesianEstimator```?

**Comment by ankurankan** [comment]
@yashu-seth Yes I meant something like that only. I should have been more clearer. The computation seems to be happening in `estimate_cpd` method. I don't think we followed any specific paper.

**Comment by lhzw** [comment]
Hi, I need this feature, then I can learn it online. Without this feature I have to calculate the count manually again and again, a little inconvenient. Any plan on this?
